src/reference/asciidoc/reactive-streams.adoc

```diff
@@ -212,12 +212,29 @@ Or in a declarative way:
 public class MainFlow {  
   @Bean
   public IntegrationFlow buildFlow() {
-     return IntegrationFlows.from(new CustomReactiveSourceMessageSource(new CustomReactiveSource()))
+     return IntegrationFlows.from(new CustomReactiveMessageProducer(new CustomReactiveSource()))
         .handle(outputChannel)
         .get();
   }
 }
 ```
+Or even without a channel adapter, we can always use the JavaDSL in the following way:
+```java
+public class MainFlow {  
+  @Bean
+  public IntegrationFlow buildFlow() {
+     return IntegrationFlows.from(this.customReactiveSource
+                .map(event - >
+                    MessageBuilder
+                    .withPayload(event.getBody())
+                    .setHeader(MyReactiveHeaders.SOURCE_NAME, event.getSourceName())
+                    .build());)
```

Comment:
The `;` is wrong here.
I would prefer to have this `Flux` extracted as a variable. It's OK to have it in this bean method definition.

Reply:
Do you mean like this?
````java
public class MainFlow {  
  @Bean
  public IntegrationFlow buildFlow() {
    Flux<Message<?>> myFlux = this.customReactiveSource
                .map(event - >
                    MessageBuilder
                    .withPayload(event.getBody())
                    .setHeader(MyReactiveHeaders.SOURCE_NAME, event.getSourceName())
                    .build());
     return IntegrationFlows.from(myFlux)
        .handle(outputChannel)
        .get();
  }
}
````

Reply:
Yep! This is good. It does give us a clue what is that `customReactiveSource` about and which `from()` factory is going to be for building an `IntegrationFlow`.